static/app/components/events/contexts/contextCard.tsx

```diff
@@ -0,0 +1,132 @@
+import {Link} from 'react-router';
+import styled from '@emotion/styled';
+
+import {
+  getContextMeta,
+  getContextTitle,
+  getFormattedContextData,
+} from 'sentry/components/events/contexts/utils';
+import {AnnotatedTextErrors} from 'sentry/components/events/meta/annotatedText/annotatedTextErrors';
+import Panel from 'sentry/components/panels/panel';
+import {StructuredData} from 'sentry/components/structuredEventData';
+import {space} from 'sentry/styles/space';
+import type {Group, Project} from 'sentry/types';
+import type {Event} from 'sentry/types/event';
+import {defined, objectIsEmpty} from 'sentry/utils';
+import useOrganization from 'sentry/utils/useOrganization';
+
+interface ContextCardProps {
+  alias: string;
+  event: Event;
+  type: string;
+  group?: Group;
+  project?: Project;
+  value?: Record<string, any>;
+}
+
+function ContextCard({alias, event, type, project, value = {}}: ContextCardProps) {
+  const organization = useOrganization();
+  if (objectIsEmpty(value)) {
+    return null;
+  }
+  const meta = getContextMeta(event, type);
+
+  const contextItems = getFormattedContextData({
+    event,
+    contextValue: value,
+    contextType: type,
+    organization,
+    project,
+  });
+
+  const content = contextItems.map(
+    ({key, subject, value: contextValue, action = {}}, i) => {
+      if (key === 'type') {
+        return null;
+      }
+      const contextMeta = meta?.[key];
+      const contextErrors = contextMeta?.['']?.err ?? [];
+      const hasErrors = contextErrors.length > 0;
+
+      const dataComponent = (
+        <StructuredData
+          value={contextValue}
+          depth={0}
+          maxDefaultDepth={0}
+          meta={contextMeta}
+          config={{}}
+          withAnnotatedText
+          withOnlyFormattedText
+        />
+      );
+
+      return (
+        <ContextContent key={i} hasErrors={hasErrors}>
+          <ContextSubject>{subject}</ContextSubject>
+          <ContextValue hasErrors={hasErrors}>
+            {defined(action?.link) ? (
+              <Link to={action.link}>{dataComponent}</Link>
+            ) : (
+              dataComponent
+            )}
+            <AnnotatedTextErrors errors={contextErrors} />
```

Comment:
I believe that this will only show errors, but not things like data scrubbing? I'd test with an event that has scrubbed values or values that are too long to see what that looks like. We still want those scrubbed values to be highlighted in some way.

Reply:
That's a good point, I will look into this before merging and add a screenshot/update code if needed. Thanks!